SFCGAL 1.3.5 fails to build with CGAL 4.12 (undefined reference to `boost::*')

As reported by Lucas Nussbaum in [Debian Bug #897487](https://bugs.debian.org/897487), SFCGAL 1.3.5 fails to build in Debian unstable where CGAL was recently updated to 4.12:
> During a rebuild of all packages in sid, your package failed to build on
amd64.
> 
> Relevant part (hopefully):
> ```
> /usr/bin/c++  -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wno-long-long -pedantic -Wpointer-arith -Wcast-align -Wcast-qual -Wno-overloaded-virtual -Wformat=2 -Winit-self -Wmissing-include-dirs -Wwrite-strings -Wno-error=undef -O2 -g -DNDEBUG  -Wl,-z,relro -Wl,-z,now -rdynamic CMakeFiles/test-regress-convex_hull.dir/main.cpp.o  -o test-regress-convex_hull -Wl,-rpath,/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/src: ../../../src/libSFCGAL.so.1.3.5 /usr/lib/x86_64-linux-gnu/libCGAL_Core.so.13.0.2 /usr/lib/x86_64-linux-gnu/libCGAL.so.13.0.2 -lgmp -lgmpxx -lmpfr -lgmp -lgmpxx -lmpfr /usr/lib/x86_64-linux-gnu/libboost_thread.so /usr/lib/x86_64-linux-gnu/libboost_chrono.so /usr/lib/x86_64-linux-gnu/libboost_system.so /usr/lib/x86_64-linux-gnu/libboost_date_time.so /usr/lib/x86_64-linux-gnu/libboost_atomic.so -lpthread 
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::typed_value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char>::xparse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) const':
> /usr/include/boost/program_options/detail/value_semantic.hpp:167: undefined reference to `boost::program_options::validate(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, int)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::basic_command_line_parser<char>::extra_parser(boost::function1<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>)':
> /usr/include/boost/program_options/detail/parsers.hpp:78: undefined reference to `boost::program_options::detail::cmdline::set_additional_parser(boost::function1<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::basic_command_line_parser<char>::run()':
> /usr/include/boost/program_options/detail/parsers.hpp:108: undefined reference to `boost::program_options::detail::cmdline::get_canonical_option_prefix()'
> /usr/include/boost/program_options/detail/parsers.hpp:109: undefined reference to `boost::program_options::detail::cmdline::run()'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::basic_command_line_parser<char>::basic_command_line_parser(int, char const* const*)':
> /usr/include/boost/program_options/detail/convert.hpp:79: undefined reference to `boost::program_options::to_internal(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::basic_command_line_parser<char>::basic_command_line_parser(int, char const* const*)':
> /usr/include/boost/program_options/detail/parsers.hpp:44: undefined reference to `boost::program_options::detail::cmdline::cmdline(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::basic_parsed_options<char> boost::program_options::parse_command_line<char>(int, char const* const*, boost::program_options::options_description const&, int, boost::function1<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&>)':
> /usr/include/boost/program_options/detail/parsers.hpp:52: undefined reference to `boost::program_options::detail::cmdline::set_options_description(boost::program_options::options_description const&)'
> /usr/include/boost/program_options/detail/parsers.hpp:70: undefined reference to `boost::program_options::detail::cmdline::style(int)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::typed_value<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char>::name() const':
> /usr/include/boost/program_options/detail/value_semantic.hpp:19: undefined reference to `boost::program_options::arg[abi:cxx11]'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::variables_map::~variables_map()':
> /usr/include/boost/program_options/variables_map.hpp:146: undefined reference to `vtable for boost::program_options::variables_map'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `main':
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:63: undefined reference to `boost::program_options::options_description::m_default_line_length'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:63: undefined reference to `boost::program_options::options_description::options_description(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned int)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:64: undefined reference to `boost::program_options::options_description::add_options()'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:65: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:66: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:67: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, char const*)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:68: undefined reference to `boost::program_options::options_description_easy_init::operator()(char const*, boost::program_options::value_semantic const*, char const*)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:71: undefined reference to `boost::program_options::variables_map::variables_map()'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:72: undefined reference to `boost::program_options::store(boost::program_options::basic_parsed_options<char> const&, boost::program_options::variables_map&, bool)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:73: undefined reference to `boost::program_options::notify(boost::program_options::variables_map&)'
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:76: undefined reference to `boost::program_options::operator<<(std::ostream&, boost::program_options::options_description const&)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `main':
> /usr/include/boost/program_options/variables_map.hpp:155: undefined reference to `boost::program_options::abstract_variables_map::operator[](std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `main':
> ./obj-x86_64-linux-gnu/test/regress/convex_hull/./test/regress/convex_hull/main.cpp:90: undefined reference to `boost::program_options::operator<<(std::ostream&, boost::program_options::options_description const&)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `main':
> /usr/include/boost/filesystem/operations.hpp:664: undefined reference to `boost::filesystem::detail::remove(boost::filesystem::path const&, boost::system::error_code*)'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o: In function `boost::program_options::variables_map::~variables_map()':
> /usr/include/boost/program_options/variables_map.hpp:146: undefined reference to `vtable for boost::program_options::variables_map'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o:(.data.rel.ro._ZTIN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE[_ZTIN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE]+0x18): undefined reference to `typeinfo for boost::program_options::value_semantic_codecvt_helper<char>'
> CMakeFiles/test-regress-convex_hull.dir/main.cpp.o:(.data.rel.ro._ZTVN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE[_ZTVN5boost15program_options11typed_valueINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEE]+0x40): undefined reference to `boost::program_options::value_semantic_codecvt_helper<char>::parse(boost::any&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_pointer_oserializer::basic_pointer_oserializer(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_pointer_iserializer'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::register_basic_serializer(boost::archive::detail::basic_oserializer const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_iprimitive<boost::archive::binary_iarchive, char, std::char_traits<char> >::init()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iserializer::basic_iserializer(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::void_upcast(boost::serialization::extended_type_info const&, boost::serialization::extended_type_info const&, void const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_iarchive>::find(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_iarchive<boost::archive::binary_iarchive>::init()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_pointer_oserializer::~basic_pointer_oserializer()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::archive_exception::archive_exception(boost::archive::archive_exception::exception_code, char const*, char const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::save_object(void const*, boost::archive::detail::basic_oserializer const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::basic_iarchive(unsigned int)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::get_library_version() const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_oprimitive<boost::archive::binary_oarchive, char, std::char_traits<char> >::save(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::load_object(void*, boost::archive::detail::basic_iserializer const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::serialization::typeid_system::extended_type_info_typeid_0'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_iserializer'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_oarchive>::find(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::extended_type_info::operator==(boost::serialization::extended_type_info const&) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_oarchive>::insert(boost::archive::detail::basic_serializer const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_iprimitive<boost::archive::binary_iarchive, char, std::char_traits<char> >::basic_binary_iprimitive(std::basic_streambuf<char, std::char_traits<char> >&, bool)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::type_unregister()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::archive_exception::~archive_exception()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_iarchive>::erase(boost::archive::detail::basic_serializer const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::type_register(std::type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_iarchive<boost::archive::binary_iarchive>::load_override(boost::archive::class_name_type&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iserializer::~basic_iserializer()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oserializer::basic_oserializer(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::~extended_type_info_typeid_0()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::extended_type_info::key_register() const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_iarchive>::insert(boost::archive::detail::basic_serializer const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_oprimitive<boost::archive::binary_oarchive, char, std::char_traits<char> >::basic_binary_oprimitive(std::basic_streambuf<char, std::char_traits<char> >&, bool)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_iarchive'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::archive_exception::archive_exception(boost::archive::archive_exception const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::save_pointer(void const*, boost::archive::detail::basic_pointer_oserializer const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::next_object_pointer(void*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_oprimitive<boost::archive::binary_oarchive, char, std::char_traits<char> >::init()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::load_pointer(void*&, boost::archive::detail::basic_pointer_iserializer const*, boost::archive::detail::basic_pointer_iserializer const* (*)(boost::serialization::extended_type_info const&))'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_oserializer'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::void_cast_detail::void_caster::recursive_unregister() const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oserializer::~basic_oserializer()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::basic_oarchive(unsigned int)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_pointer_iserializer::~basic_pointer_iserializer()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::end_preamble()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::get_extended_type_info(std::type_info const&) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_pointer_iserializer::basic_pointer_iserializer(boost::serialization::extended_type_info const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::archive_exception'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::text_iarchive_impl<boost::archive::text_iarchive>::load(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::~basic_iarchive()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_iprimitive<boost::archive::binary_iarchive, char, std::char_traits<char> >::~basic_binary_iprimitive()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_oarchive'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::extended_type_info::key_unregister() const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `typeinfo for boost::archive::detail::basic_pointer_oserializer'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::void_cast_detail::void_caster::recursive_register(bool) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_oarchive<boost::archive::binary_oarchive>::init()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::is_less_than(boost::serialization::extended_type_info const&) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::is_equal(boost::serialization::extended_type_info const&) const'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_iarchive::register_basic_serializer(boost::archive::detail::basic_iserializer const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::text_oarchive_impl<boost::archive::text_oarchive>::save(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::void_downcast(boost::serialization::extended_type_info const&, boost::serialization::extended_type_info const&, void const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::serialization::typeid_system::extended_type_info_typeid_0::extended_type_info_typeid_0(char const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::basic_oarchive::~basic_oarchive()'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::detail::archive_serializer_map<boost::archive::binary_oarchive>::erase(boost::archive::detail::basic_serializer const*)'
> ../../../src/libSFCGAL.so.1.3.5: undefined reference to `boost::archive::basic_binary_oprimitive<boost::archive::binary_oarchive, char, std::char_traits<char> >::~basic_binary_oprimitive()'
> collect2: error: ld returned 1 exit status
> ```
> The full build log is available from:
>    http://aws-logs.debian.net/2018/05/02/sfcgal_1.3.5-1_unstable.log
> 
> A list of current common problems and possible solutions is available at
> http://wiki.debian.org/qa.debian.org/FTBFS . You're welcome to contribute!
> 
> About the archive rebuild: The rebuild was done on EC2 VM instances from
> Amazon Web Services, using a clean, minimal and up-to-date chroot. Every
> failed build was retried once to eliminate random failures.

CC: @sloriot

--- a/example/CGAL-basic_manip/CMakeLists.txt
+++ b/example/CGAL-basic_manip/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/CGAL-cartesian_kernel/CMakeLists.txt
+++ b/example/CGAL-cartesian_kernel/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/CGAL-point_generator/CMakeLists.txt
+++ b/example/CGAL-point_generator/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/CGAL-polygon_triangulation2/CMakeLists.txt
+++ b/example/CGAL-polygon_triangulation2/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/CGAL-triangulation2/CMakeLists.txt
+++ b/example/CGAL-triangulation2/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/SFCGAL-building/CMakeLists.txt
+++ b/example/SFCGAL-building/CMakeLists.txt
@@ -2,6 +2,8 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL SFCGAL-osg)
+
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES} SFCGAL-osg)
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/SFCGAL-export-osg/CMakeLists.txt
+++ b/example/SFCGAL-export-osg/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL SFCGAL-osg)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL SFCGAL-osg ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/example/SFCGAL-offset/CMakeLists.txt
+++ b/example/SFCGAL-offset/CMakeLists.txt
@@ -2,6 +2,7 @@ GET_FILENAME_COMPONENT( EXAMPLE_NAME ${CMAKE_CURRENT_SOURCE_DIR} NAME )
 add_executable( example-${EXAMPLE_NAME} 
 	main.cpp 
 )
-target_link_libraries( example-${EXAMPLE_NAME} SFCGAL)
+find_package(Boost REQUIRED COMPONENTS serialization)
+target_link_libraries( example-${EXAMPLE_NAME} SFCGAL ${Boost_LIBRARIES})
 set_target_properties( example-${EXAMPLE_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS example-${EXAMPLE_NAME} DESTINATION bin )

--- a/test/bench/CMakeLists.txt
+++ b/test/bench/CMakeLists.txt
@@ -1,6 +1,9 @@
 file( GLOB SFCGAL_BENCHS_SOURCES *.cpp )
 add_executable( bench-SFCGAL ${SFCGAL_BENCHS_SOURCES} )
-target_link_libraries( bench-SFCGAL SFCGAL)
+
+find_package(Boost REQUIRED COMPONENTS unit_test_framework timer serialization)
+
+target_link_libraries( bench-SFCGAL SFCGAL ${Boost_LIBRARIES})
 set_target_properties( bench-SFCGAL PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS bench-SFCGAL DESTINATION bin )
 

--- a/test/garden/CMakeLists.txt
+++ b/test/garden/CMakeLists.txt
@@ -1,11 +1,13 @@
 #-- polygon_triangulator test
 file( GLOB_RECURSE SFCGAL_REGRESS_GARDEN_TEST_SOURCES *.cpp )
 
+find_package(Boost REQUIRED COMPONENTS program_options serialization)
+
 set( REGRESS_NAME garden-test-SFCGAL )
 add_executable( ${REGRESS_NAME} ${SFCGAL_REGRESS_GARDEN_TEST_SOURCES} )
 
 target_link_libraries( ${REGRESS_NAME} SFCGAL)
-target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES} )
+target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES} ${Boost_LIBRARIES})
 
 set_target_properties( ${REGRESS_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS ${REGRESS_NAME} DESTINATION bin )

--- a/test/regress/convex_hull/CMakeLists.txt
+++ b/test/regress/convex_hull/CMakeLists.txt
@@ -4,9 +4,11 @@ file( GLOB_RECURSE SFCGAL_REGRESS_CONVEX_HULL_TEST_SOURCES *.cpp )
 set( REGRESS_NAME test-regress-convex_hull )
 add_executable( ${REGRESS_NAME} ${SFCGAL_REGRESS_CONVEX_HULL_TEST_SOURCES} )
 
+find_package(Boost REQUIRED COMPONENTS program_options filesystem serialization)
+
 target_link_libraries( ${REGRESS_NAME} SFCGAL)
 
-target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES} )
+target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES} ${Boost_LIBRARIES})
 
 set_target_properties( ${REGRESS_NAME} PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS ${REGRESS_NAME} DESTINATION bin )

--- a/test/regress/polygon_triangulator/CMakeLists.txt
+++ b/test/regress/polygon_triangulator/CMakeLists.txt
@@ -4,8 +4,10 @@ file( GLOB_RECURSE SFCGAL_REGRESS_POLYGON_TRIANGULATOR_TEST_SOURCES *.cpp )
 set( REGRESS_NAME test-regress-polygon_triangulator )
 add_executable( ${REGRESS_NAME} ${SFCGAL_REGRESS_POLYGON_TRIANGULATOR_TEST_SOURCES} )
 
+find_package(Boost REQUIRED COMPONENTS program_options filesystem serialization)
+
 target_link_libraries( ${REGRESS_NAME}	SFCGAL)
-target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES})
+target_link_libraries( ${REGRESS_NAME} ${CGAL_3RD_PARTY_LIBRARIES} ${Boost_LIBRARIES})
 
 
 set_target_properties( ${REGRESS_NAME} PROPERTIES DEBUG_POSTFIX "d" )

--- a/test/regress/standalone/CMakeLists.txt
+++ b/test/regress/standalone/CMakeLists.txt
@@ -2,7 +2,9 @@
 file( GLOB_RECURSE SFCGAL_REGRESS_STANDALONE_TEST_SOURCES *.cpp )
 add_executable( standalone-regress-test-SFCGAL ${SFCGAL_REGRESS_STANDALONE_TEST_SOURCES} )
 
-target_link_libraries( standalone-regress-test-SFCGAL SFCGAL)
+find_package(Boost REQUIRED COMPONENTS unit_test_framework filesystem serialization)
+
+target_link_libraries( standalone-regress-test-SFCGAL SFCGAL ${Boost_LIBRARIES})
 
 set_target_properties( standalone-regress-test-SFCGAL PROPERTIES DEBUG_POSTFIX "d" )
 install( TARGETS standalone-regress-test-SFCGAL DESTINATION bin )

--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -1,8 +1,11 @@
 #-- build unit tests
+
+find_package(Boost REQUIRED COMPONENTS unit_test_framework serialization)
+
 file( GLOB_RECURSE SFCGAL_UNIT_TEST_SOURCES *.cpp )
 add_executable( unit-test-SFCGAL ${SFCGAL_UNIT_TEST_SOURCES} )
 target_link_libraries( unit-test-SFCGAL SFCGAL)
-target_link_libraries(unit-test-SFCGAL ${CGAL_3RD_PARTY_LIBRARIES})
+target_link_libraries(unit-test-SFCGAL ${CGAL_3RD_PARTY_LIBRARIES} ${Boost_LIBRARIES})
 
 #include( PrecompiledHeader )
 #if(PCHSupport_FOUND)

